packages/deploy-tooling/src/base/deploy.ts

```diff
@@ -281,6 +326,47 @@ async function getAbapService(config: AbapDeployConfig, logger: Logger): Promise
     return service;
 }
 
+/**
+ * Creates a new transport request using adt service.
+ *
+ * @param config - deployment configuration
+ * @param logger - reference to the logger instance
+ * @param provider - instance of the axios-extension abap service provider (it is created if not passed in)
+ * @throws error if transport request creation fails
+ * @returns transportRequest
+ */
+export async function createTransportRequest(
+    config: AbapDeployConfig,
+    logger: Logger,
+    provider?: AbapServiceProvider
+): Promise<string | void> {
+    if (!provider) {
+        provider = await getAbapServiceProvider(config, logger);
+    }
+
+    const adtService = await provider.getAdtService<TransportRequestService>(TransportRequestService);
+    if (adtService) {
```

Comment:
Return type should be Promise<string> instead of Promise<string|void>

You have the void case because the full execution logic is in `if (adtService) {...}`. If this branch is not executed, void type is expected. IMO, you can throw error when adtService is not available, because it is the same scenario as create transport request cannot be succeeded and the subsequent deploy cannot continue.

Reply:
@slin-sap what is the reason why `getAdtService` would return null? Surely that should throw an exception if not found? As you cant continue if not found.

Reply:
That should be handled now.